python/pacemaker/_cts/test.py

```diff
@@ -262,28 +265,34 @@ def _start_daemons(self):
     # PUBLIC METHODS
     #
 
-    def add_cmd(self, cmd, args, validate=True, check_rng=True, check_stderr=True,
-                env=None):
+    def add_cmd(self, cmd=None, **kwargs):
         """Add a simple command to be executed as part of this test."""
-        self._new_cmd(cmd, args, ExitStatus.OK, validate=validate, check_rng=check_rng,
-                      check_stderr=check_stderr, env=env)
+        self._new_cmd(cmd, kwargs.pop("args", ""), ExitStatus.OK,
+                      validate=kwargs.get("validate", True),
+                      check_rng=kwargs.get("check_rng", True),
+                      check_stderr=kwargs.get("check_stderr", True),
+                      env=kwargs.get("env"))
 
-    def add_cmd_and_kill(self, cmd, args, kill_proc):
+    def add_cmd_and_kill(self, cmd=None, **kwargs):
         """Add a command and system command to be executed as part of this test."""
-        self._new_cmd(cmd, args, ExitStatus.OK, kill=kill_proc)
+        self._new_cmd(cmd, kwargs.pop("args", ""), ExitStatus.OK,
+                      kill=kwargs.get("kill"))
```

Comment:
Do we even need the extra add methods anymore? If we're passing everything as named arguments in the callers, they can just always use `add_cmd`

Don't necessarily need to do anything about it in this PR though

Reply:
We could probably move the contents of Test._new_cmd into Test.add_cmd, remove all the other Test.add_* functions, change ExecTest.add_cmd to deal with the default program name, and get rid of all the other ExecTest.add_* functions.  It would be a lot of caller changing.